core/chain/coin/knownTokens/resolve.ts

```
@@ -0,0 +1,98 @@
+    fetchJson<{ metadata?: DenomMetadata }>(byDenom)
+  )
+  if (byDenomRes.data?.metadata) return byDenomRes.data.metadata
+
+  // Fallback: list & filter
+  const listUrl = `${lcdBase}/cosmos/bank/v1beta1/denoms_metadata?pagination.limit=1000`
+  const listRes = await attempt(() =>
+    fetchJson<{ metadatas?: DenomMetadata[] }>(listUrl)
+  )
+  return listRes.data?.metadatas?.find(data => data.base === denom) ?? null
+
+  return null
```

Biome (2.1.2)

[error] This code will never be reached ...

... because this statement will return from the function beforehand


(lint/correctness/noUnreachable)